Annotations/Classes/Views/ArrowCanvas.swift

```diff
@@ -14,25 +14,27 @@ protocol ArrowCanvas: class, ArrowViewDelegate {
 }
 
 extension ArrowCanvas {
-  func redrawArrows(model: CanvasModel) {
-    for (index, model) in model.arrows.enumerated() {
-      let state = ArrowViewState(model: model, isSelected: false)
-      let view = ArrowViewClass(state: state, modelIndex: index, color: model.color)
-      view.delegate = self
-      add(view)
-    }
+  func redrawArrow(model: ArrowModel, canvas: CanvasModel) {
+    guard let modelIndex = canvas.arrows.firstIndex(of: model) else { return }
+    let state = ArrowViewState(model: model, isSelected: false)
+    let view = ArrowViewClass(state: state, modelIndex: modelIndex,
+                              globalIndex: model.index, color: model.color)
+    view.delegate = self
+    add(view)
   }
   
   func createArrowView(origin: PointModel, to: PointModel, color: ModelColor) -> (CanvasDrawable?, KnobView?) {
     if origin.distanceTo(to) < 10 {
       return (nil, nil)
     }
     
-    let newArrow = ArrowModel(origin: origin, to: to, color: color)
+    let newArrow = ArrowModel(index: model.elements.count + 1,
+                              origin: origin, to: to, color: color)
```

Comment:
@dworak for better readability it is worth starting each function parameter from a new row:
ArrowModel(index:. ..
                     origin: ...,
                     to: ...,
                     color: ... )